### PR TITLE
[SYCL] Add BFloat16 feature end-to-end test

### PR DESCRIPTION
**Pull request by MrSidims**
Spec: https://github.com/intel/llvm/pull/4237
Implementation: https://github.com/intel/llvm/pull/4213

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: SYCL/BFloat16/bfloat16_type.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      cl::sycl::ext::intel::experimental::bfloat16 BVal{B[index]};
+      cl::sycl::ext::intel::experimental::bfloat16 CVal = AVal + BVal;
+      C[index] = CVal;
+    });
+  });
+
+  assert_close(c.get_access<access::mode::read>(), ref);
+}
+
+void verify_min(queue &q, buffer<float, 1> &a, buffer<float, 1> &b, range<1> &r,
```
Comment (AlexeySotkin) [inline SYCL/BFloat16/bfloat16_type.cpp]:
```suggestion
void verify_sub(queue &q, buffer<float, 1> &a, buffer<float, 1> &b, range<1> &r,
```

File: SYCL/BFloat16/bfloat16_type.cpp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  assert_close(a.get_access<access::mode::read>(), ref);
+}
+
+void verify_conv_explicit(queue &q, buffer<float, 1> &a, range<1> &r,
+                          const float ref) {
+  q.submit([&](handler &cgh) {
+    auto A = a.get_access<access::mode::read_write>(cgh);
+    cgh.parallel_for<class calc_conv>(r, [=](id<1> index) {
+      cl::sycl::ext::intel::experimental::bfloat16 AVal = from_float(A[index]);
+      A[index] = to_float(AVal);
```
Comment (AlexeySotkin) [inline SYCL/BFloat16/bfloat16_type.cpp]: Workflow for explicit conversion is expected to be like :
```suggestion
      uint16_t AVal = bfloat16::from_float(A[index]);
      A[index] = bfloat16::to_float(AVal);
```

File: SYCL/BFloat16/bfloat16_type.cpp [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  });
+
+  assert_close(c.get_access<access::mode::read>(), ref);
+}
+
+int main() {
+  device dev{default_selector()};
+
+  if (!dev.has(aspect::ext_intel_bf16_conversio()) {
+    std::cout << "This device doesn't support bfloat16 type" << std::endl;
```
Comment (AlexeySotkin) [inline SYCL/BFloat16/bfloat16_type.cpp]:
```suggestion
    std::cout << "This device doesn't support bfloat16 conversion feature" << std::endl;
```

File: SYCL/BFloat16/bfloat16_type.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      if (AVal && !BVal) {
+        if (AVal > BVal || AVal >= BVal || AVal < BVal || AVal <= BVal)
+          cl::sycl::ext::intel::experimental::bfloat16 CVal =
+              AVal != BVal ? AVal : BVal;
+          CVal--;
+          CVal++;
+          if (AVal == BVal)
+            CVal += AVal;
+          else
+            CVal += BVal;
```
Comment (AlexeySotkin) [inline SYCL/BFloat16/bfloat16_type.cpp:161]: `+=` is used in both branches of the `if ... else` statement. I believe one of them should be `-=`, or `*=` or `/=`.

Comment (MrSidims) [reply SYCL/BFloat16/bfloat16_type.cpp:161]: Applied

File: SYCL/BFloat16/bfloat16_type.cpp (hunk 3 above)
Comment (MrSidims) [reply SYCL/BFloat16/bfloat16_type.cpp]: thanks!

File: SYCL/BFloat16/bfloat16_type.cpp (hunk 2 above)
Comment (MrSidims) [reply SYCL/BFloat16/bfloat16_type.cpp]: thanks!

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
Could you please define code owners for the directory in .github/CODEOWNERS
I think @AlexeySotkin and @MrSidims are the best candidates.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**